libraries/core/src/androidTest/kotlin/com/bumble/appyx/core/node/PermanentChildTest.kt

```diff
@@ -11,29 +11,43 @@ import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.test.hasTestTag
 import com.bumble.appyx.core.AppyxTestScenario
 import com.bumble.appyx.core.children.nodeOrNull
+import com.bumble.appyx.core.composable.PermanentChild
 import com.bumble.appyx.core.modality.BuildContext
-import com.bumble.appyx.core.navigation.EmptyNavModel
+import com.bumble.appyx.core.navigation.model.permanent.PermanentNavModel
 import kotlinx.parcelize.Parcelize
 import org.junit.Assert.assertEquals
 import org.junit.Rule
 import org.junit.Test
 
 class PermanentChildTest {
 
+    var nodeFactory: (buildContext: BuildContext) -> TestParentNode = {
+        TestParentNode(buildContext = it)
+    }
+
     @get:Rule
     val rule = AppyxTestScenario { buildContext ->
-        TestParentNode(buildContext)
+        nodeFactory(buildContext)
     }
 
     @Test
-    fun permanent_child_is_rendered() {
+    fun `WHEN_permanent_model_contains_relevant_nav_key_THEN_permanent_child_is_rendered`() {
+        createPermanentNavModelWithNavKey()
         rule.start()
 
         rule.onNode(hasTestTag(TestParentNode.NavTarget::class.java.name)).assertExists()
     }
 
     @Test
-    fun permanent_child_is_reused_when_visibility_switched() {
```

Comment:
Can you re-add this test (without createPermanentNavModelWithNavKey)? 
This would verify that the late creation of the child continues to work

Reply:
The test is still there, i just renamed it to `WHEN_visibility_switched_THEN_permanent_child_is_reused`